### PR TITLE
[Fix][TTS] Rebuild Qwen3-TTS and delay Moss re-prefill after retraction

### PR DESCRIPTION
**Pull request by rwang5203**
## Motivation

After KV-pressure retraction, the scheduler re-prefills the prompt plus already-generated tokens. Qwen3-TTS and delay Moss previously sliced prompt-only projected buffers, so the runners could provide fewer embedding rows than the scheduler's extend length and KV allocation.

Reconstructed generated tails also need radix isolation. Both runners publish scalar scheduler IDs that do not fully identify the audio-conditioned embeddings written to KV.

## Modifications

- Rebuild Qwen3-TTS re-prefill inputs from the prompt plus recorded decode-input history, including the final queued feedback row.
- Rebuild delay-Moss inputs from `prompt_rows + output_rows` and clear feedback stranded by retraction after validating the reconstructed row count.
- Give every Qwen3-TTS and delay-Moss request lifetime a unique radix `extra_key`.
- Add focused Qwen cadence, Moss feedback-lifecycle, failure-ordering, and radix-isolation regressions.

## Related Issues

Addresses part of https://github.com/sgl-project/sglang-omni/issues/1555. Zonos2 retraction recovery remains out of scope, so this PR does not close that issue.

## Accuracy Test

```bash
pytest -q -rs tests/unit_test/moss_tts tests/unit_test/qwen3_tts
```

Result: `349 passed, 2 skipped`. Both skips require unavailable `flash_attn_varlen_func` and are unrelated to this change.

The regressions cover the observed Qwen `594`-vs-`460` mismatch, production feedback cadence, Moss reconstruction and stale-feedback lifecycle, row-mismatch failure ordering, and request-lifetime radix isolation. No model weights, logits, or sampling math change.

## Benchmark & Profiling

No benchmark was run. The request-lifetime `extra_key` disables cross-request radix-prefix sharing for Qwen3-TTS and delay Moss, while preserving same-request cache reuse across retractions. That is the conservative correctness fence. Content-addressed generated keys can recover safe cross-request sharing in a follow-up.

### CODE REVIEW

**Comment by charliechenye** [comment]
how much cross-request radix hit rate / TTFT changes because of the unique extra_key.

**Comment by rwang5203** [comment]
Reran unique `extra_key` versus `extra_key=None` on one H200 per server for Qwen3-TTS-12Hz-0.6B-Base and MOSS-TTS-v1.5 delay at concurrency 1, 8, and 16.

With a unique `extra_key`, every request's initial cross-request cache match was **0**, including concurrent waves. With `extra_key=None`:

- identical repeats cached 58/59 tokens for Qwen and 148/149 for Moss
- same-voice different-text requests typically cached 24/59 for Qwen, and 131/149 on the representative Moss request
- concurrent identical waves still cached 58 tokens per Qwen request and 148 per Moss request
- long identical repeats cached 1181/1182 for Qwen and 1271/1272 for Moss

At concurrency 1, median TTFT differed by at most 2 ms. At concurrency 8 with the 94 s reference (unique minus `None`), Qwen ranged from -2 to +202 ms in the first wave and +23 to +33 ms in the second, depending on warmup and arm order. Moss stayed at **+148 to +211 ms** across both waves and arm orders.

TTFT is the first Qwen vocoder burst or the first Moss PCM frame.

File: sglang_omni/models/moss_tts/model_runner.py [hunk 1]
```diff
@@ -88,7 +88,21 @@ def _build_prefill_input_embeds(
                 raise RuntimeError("MOSS-TTS prefill requires prompt_rows")
             req_len = int(req.extend_range.length)
             prefix_len = len(req.prefix_indices)
+            if data.output_rows:
+                # Retraction re-prefill includes generated frames; drop the
```
Comment (luojiaxuan) [inline sglang_omni/models/moss_tts/model_runner.py:92]: add notes xxx

File: tests/unit_test/moss_tts/test_pipeline.py [hunk 2]
```diff
@@ -1299,6 +1302,99 @@ def encode(self, path: str) -> torch.Tensor:
...
+    assert first.req.extra_key != second.req.extra_key
+    assert list(first.req.origin_input_ids) == list(second.req.origin_input_ids)
+
+    kept = first.req.extra_key
+    first.req.reset_for_retract()
+    assert first.req.extra_key == kept
+
+
+def test_moss_lifetime_extra_key_isolates_delay_slot_generated_prefix() -> None:
+    """Generated radix keys are text-channel IDs (rows[:, 0]). Continuing audio
```
Comment (luojiaxuan) [inline tests/unit_test/moss_tts/test_pipeline.py:1368]: add notes xxx

File: tests/unit_test/moss_tts/test_pipeline.py [hunk 3]
```diff
@@ -1587,6 +1684,128 @@ def _prepare_multi_modal_inputs(self, rows):
...
+
+    with pytest.raises(RuntimeError, match="prefill row mismatch"):
+        _retract_runner()._build_prefill_input_embeds(forward_batch, [sched_req])
+
+    assert len(queue) == 1
+    assert torch.equal(queue[0], stranded)
+
+
+def test_moss_reprefill_discards_stranded_feedback() -> None:
+    """Retraction resume must consume the newly sampled frame, not the old queue row."""
```
Comment (luojiaxuan) [inline tests/unit_test/moss_tts/test_pipeline.py:1784]: add notes prefix

File: tests/unit_test/qwen3_tts/test_retract_prefill.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        torch.tensor([21.0, 32.0]),
+    )
+    assert torch.equal(embedding.weight[0].detach(), torch.tensor([21.0, 32.0]))
+    assert forward_batch.input_ids.tolist() == [0]
+    assert list(sched_req.data.pending_feedback_queue) == []
+    assert list(sched_req.data.pending_text_queue) == []
+
+
+def test_reprefill_after_retract_replays_prompt_plus_generated() -> None:
+    """Production cadence: G generated tokens, history G-1, one queued row.
```
Comment (luojiaxuan) [inline tests/unit_test/qwen3_tts/test_retract_prefill.py:87]: add notes xxx

**Review by luojiaxuan — APPROVED** [review]
add notes prefix and others LGTM
